From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 16 Apr 2021 14:01:58 -0400
Subject: [PATCH] Don't add build_root/giscanner to PYTHONPATH for running
 tests.

When testing a PyPy build on macOS, the 'ast.py' module in giscanner
shadows the Python standard library ast module, resuting in an import
error. This removes the extra giscanner path, modifies the
environment for running 'warningtester.py' so that it will do python
imports from the build dir, and modifies 'warningtester.py' so that it
doesn't insert the source dir in sys.path.
---
 tests/warn/meson.build      | 7 ++-----
 tests/warn/warningtester.py | 5 ++---
 2 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/tests/warn/meson.build b/tests/warn/meson.build
index 9641787f..597afa7b 100644
--- a/tests/warn/meson.build
+++ b/tests/warn/meson.build
@@ -21,12 +21,9 @@ warn_tests = [
 ]
 
 warn_test_env = environment()
-warn_test_env.set('UNINSTALLED_INTROSPECTION_SRCDIR', source_root)
+warn_test_env.set('TOP_SRCDIR', source_root)
 warn_test_env.set('TOP_BUILDDIR', build_root)
-warn_test_env.set(
-  'PYTHONPATH',
-  build_root,
-  join_paths(build_root, 'giscanner'))
+warn_test_env.set('PYTHONPATH', build_root)
 
 # FIXME: Glib as a subproject (used on Windows mostly).
 if glib_dep.type_name() == 'pkgconfig'
diff --git a/tests/warn/warningtester.py b/tests/warn/warningtester.py
index 3bb9e4b2..ccec7b6c 100644
--- a/tests/warn/warningtester.py
+++ b/tests/warn/warningtester.py
@@ -3,9 +3,8 @@ import sys
 import difflib
 import builtins
 
-path = os.getenv('UNINSTALLED_INTROSPECTION_SRCDIR', None)
+path = os.getenv('TOP_SRCDIR', None)
 assert path is not None
-sys.path.insert(0, path)
 
 # Not correct, but enough to get the tests going uninstalled
 builtins.__dict__['DATADIR'] = path
@@ -24,7 +23,7 @@ from giscanner.scannermain import process_packages
 currentdir = os.path.dirname(os.path.abspath(sys.argv[0]))
 current_name = os.path.basename(currentdir)
 path = os.path.abspath(os.path.join(currentdir, '..', ''))
-top_srcdir = os.environ['UNINSTALLED_INTROSPECTION_SRCDIR']
+top_srcdir = os.environ['TOP_SRCDIR']
 top_builddir = os.environ['TOP_BUILDDIR']
 
 
-- 
2.35.0

